### PR TITLE
Handle additional edge cases for MRTK inspectors

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview

There are two key dependencies between profiles/inputhandlers: input actions and speech keywords (i.e speechinputhandler). The inspector logic for these impacted components has been updated to be simpler, more consistent, and accurate with useful error messages to the user informing them of the issue (i.e no MRTK in scene, etc etc.). 

Rendering of speech commands when rendered as sub-profile view (i.e within input profile) would not work properly for layout. Switched model to be vertical components to preserve useful formatting

Added forced height on scrollview for input actions inspector when rendered as sub-profile

Fixed edge case where inspector logic does not account for an active profile assigned to the scene MRTK instance (an unlikely edge case)

Fixed DefaultMouseProfile to correctly set IsCustomProfile to false

Fixed DefaultHololens2..profiles to correctly set IsCustomProfile to false

Fixed "Back to.." buttons for profile up the hierarchy to actually render

## Changes
- Fixes: #4520 SpeechInputHandlerInspector broken
#4467 
Harden MRTK inspectors & increase re-usability of code 
#4654 
Can't bind the keyboard input for voice input

## Verification

Removed input actions profile
Removed active profile
Removed MRTK instance
Switched hierarchy with multiple gesture profiles (but only one active in instance)
Deleted MRTK instance & re-added with SpeechInputHandler button when error message shows

### CODE REVIEW

**Comment by Troy-Ferrell** [comment]
@luis-valverde-ms and @provencher FYI

File: Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs [hunk 1]
```diff
@@ -23,5 +23,32 @@ public override void OnInspectorGUI()
...
         }
+
+        /// <summary>
+        /// If MRTK is not initialized in scene, then show an error message and button to add MRTK to the scene
+        /// If MRTK does not has an active profile, show an error message
+        /// </summary>
+        /// <returns>true if MRTK is initialized and has active profile, false otherwise</returns>
+        protected bool CheckMixedRealityToolkit()
+        {
+            if (!MixedRealityToolkit.IsInitialized)
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs]: A MixedRealityToolkit, instance is no longer required to use the MRTK (this is in the process of becoming easier to configure).

This method should likely not be added.

File: Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/PointerClickHandlerInspector.cs [hunk 2]
```diff
@@ -26,13 +27,35 @@ public override void OnInspectorGUI()
         {
             base.OnInspectorGUI();
 
-            if (!MixedRealityInspectorUtility.CheckMixedRealityConfigured(false)) { return; }
+            bool enabled = CheckMixedRealityToolkit();
+            if (enabled)
+            {
+                if (!MixedRealityToolkit.Instance.ActiveProfile.IsInputSystemEnabled)
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/PointerClickHandlerInspector.cs]: The new pattern for accessing profiles and systems has changed. Please see #4315 and #4437 for reference and feel free to reach out with questions.

Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/PointerClickHandlerInspector.cs]: @davidkline-ms , hmm so what service then contains IsInputSystemEnabled as your commet highlighted(aka what service consumes MixedRealityToolkitConfigurationProfile in it's constructor)? The only method I see for accessing profiles is roughly:

```
IMixedRealityService service;
MixedRealityServiceRegistry.TryGetService<IMixedRealityService>(out service); 

svc.profile.blah.dosomething etc
```

File: Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs (hunk 1 above)
Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs]: @davidkline-ms if there is no MRTK instance in scene, how does one get the list of available inputactions? Looking at the updated Interactable.cs, it still checks if MRTK is initialized. 

       ```
 public static bool TryGetInputActions(out string[] descriptionsArray)
        {
            if (!MixedRealityToolkit.ConfirmInitialized() || !MixedRealityToolkit.Instance.HasActiveProfile)
            {
                descriptionsArray = null;
                return false;
            }

            MixedRealityInputAction[] actions = InputSystem.InputSystemProfile.InputActionsProfile.InputActions;

            descriptionsArray = new string[actions.Length];
            for (int i = 0; i < actions.Length; i++)
            {
                descriptionsArray[i] = actions[i].Description;
            }

            return true;
        }
```

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs]: The code snippet above uses a cached input system that was acquired using

```
MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>()
```

This pattern is supported for the MixedRealityToolkit object as well as standalone services.

Please let me know if you need more clarity or have questions.

Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs]: @davidkline-ms , yes it uses a cached input system. The point though is that the input actions list is only grabbed from the MRTK instance currently in scene. If there is no MRTK instance in scene, then you can't get the input actions list. Thus, my CheckMixedRealityToolkit() is still necessary. There is still a dependency

File: Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/PointerClickHandlerInspector.cs (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/PointerClickHandlerInspector.cs]: at this point, there is nothing that consumes that profile, other than the MixedRealityToolkit object.

based on feedback from some customers, there will be scenarios where there isnt a MixedRealityToolkit object in the scene.

many (though not yet all) instances of Is<Service>Enabled are replaceable with a check to see if the system is non-null

File: Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.SDK/Inspectors/Input/Handlers/BaseInputHandlerInspector.cs]: The primary concern is to not add any new references to MixedRealityToolkit.Instance. Let's discuss and post the results here.

**Comment by Troy-Ferrell** [comment]
After syncing with @davidkline-ms , the use of MRTK object references are going to remain in the PR. This is because 
1) These are only editor code additions, not runtime use additions
2) The TryGetService & related paradigm needs to be rationalized for editor use since MRTK is not actually running

**Comment by wiwei** [comment]
Thanks!

**Pull request merged**